src/CentrifugeRouter.sol

```diff
@@ -9,6 +9,11 @@ import {IERC20} from "src/interfaces/IERC20.sol";
 import {ICentrifugeRouter} from "src/interfaces/ICentrifugeRouter.sol";
 import {IPoolManager} from "src/interfaces/IPoolManager.sol";
 
+interface GatewayLike {
+    function topUp() external payable;
+    function estimate(bytes calldata payload) external returns (uint256[] memory tranches, uint256 total);
```

Comment:
I would combine topup and estimate to one function call that:
- returns the gas estimation
- sets the prepaid amount / fuel balance for prepaid tx